src/vcpkg/commands.export-port.cpp

```diff
@@ -0,0 +1,216 @@
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
+        for (const auto& file : Util::fmap(port_files, [&source](StringView&& str) -> Path {
+                 return str.substr(source.generic_u8string().size() + 1);
+             }))
+        {
+            const auto src_path = source / file;
+            const auto dst_path = destination / file;
+            fs.create_directories(dst_path.parent_path(), IgnoreErrors{});
+            fs.copy_file(src_path, dst_path, CopyOptions::overwrite_existing, VCPKG_LINE_INFO);
+        }
+    }
+
+    void export_registry_port(const VcpkgPaths& paths,
+                              StringView port_name,
+                              Optional<Version> version,
+                              const Path& destination)
+    {
+        // Registry configuration
+        const auto& config = paths.get_configuration();
+        auto registries = config.instantiate_registry_set(paths);
+        auto source = version ? registries->fetch_port_files(port_name, version.value_or_exit(VCPKG_LINE_INFO))
+                              : registries->fetch_port_files(port_name);
+        if (!source)
+        {
+            msg::println_error(source.error());
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+        copy_port_files(paths.get_filesystem(), port_name, source.value_or_exit(VCPKG_LINE_INFO).path, destination);
+        msg::println(msgExportPortSuccess, msg::path = destination);
+        Checks::exit_success(VCPKG_LINE_INFO);
+    }
+
+    void export_classic_mode_versioned(const VcpkgPaths& paths,
+                                       StringView port_name,
+                                       const Version& version,
+                                       const Path& destination)
+    {
+        const auto db_file = paths.builtin_registry_versions / fmt::format("{}-/{}.json", port_name[0], port_name);
+
+        auto& fs = paths.get_filesystem();
+        if (!fs.exists(db_file, VCPKG_LINE_INFO))
+        {
+            msg::println_error(msgExportPortVersionsDbFileMissing, msg::package_name = port_name, msg::path = db_file);
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
+        auto contents = fs.read_contents(db_file, VCPKG_LINE_INFO);
+        auto maybe_db = parse_git_versions_file(contents, db_file);
+        if (!maybe_db)
+        {
+            msg::println_error(maybe_db.error());
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
+        auto db = std::move(maybe_db.value_or_exit(VCPKG_LINE_INFO));
+        for (auto&& entry : db)
+        {
+            if (entry.version == version)
+            {
+                const Path parent_dir = destination.parent_path();
+                fs.create_directories(parent_dir, VCPKG_LINE_INFO);
+
+                const auto archive_path =
+                    parent_dir /
+                    fmt::format("{}-{}-{}.tar", port_name, entry.version.text(), entry.version.port_version());
```

Comment:
Version strings can include special characters not allowed in name files. I think using the first 10 hexits from the version hash would be better.